lib/sink.js

```diff
@@ -73,61 +71,95 @@ class WriteStream extends stream.PassThrough {
 }
 
 class ReadStream extends stream.PassThrough {
-    constructor(gcs, bucket, file, options) {
+    constructor(gcsFile, fileName, options) {
         super();
-        const gcsFile = bucket.file(file);
         const gcsStream = gcsFile.createReadStream(options);
 
         gcsStream.on('response', () => {
-            this.emit('file found', file);
+            this.emit('file found', fileName);
         });
 
         gcsStream.on('error', () => {
-            this.emit('file not found', file);
+            this.emit('file not found', fileName);
         });
 
         gcsStream.pipe(this);
     }
 }
 
 module.exports = class SinkGCS extends EventEmitter {
-    constructor(options, bucket, metadata = {}) {
+    constructor(options, bucketName, metadata = {}) {
         super();
         assert(
             options && isObject(options),
             '"options" object must be provided'
         );
-        assert(bucket && isString(bucket), '"bucket" string must be provided');
+        assert(
+            bucketName && isString(bucketName),
+            '"bucket" string must be provided'
+        );
 
         this.options = options;
+        this.bucketName = bucketName;
         this.name = 'asset-pipe-sink-gcs';
         this.gcs = gcsClient(options);
         this.bucket = null;
 
-        this.gcs.createBucket(bucket, metadata, (error, bucketObj) => {
-            if (error) {
-                this.bucket = this.gcs.bucket(bucket);
-                return this.emit(
+        this.ready = this.gcs
```

Comment:
Is this intended as an external api change? If it is should this be documented in the readme? 
Do the other sinks also have a `.ready`? If not, maybe we should be being consistent with it?

Reply:
Thx - Yeah, this needs to be documented. Looks like in the code that we need a ready call....

Reply:
I could make it "private" for now? If we want to delay this.

Reply:
how would you make it private? move the async work into the individual methods?

Reply:
this._ready, as in javascript "'"private"'".

Reply:
Yeah that might be a nice way to go.

Reply:
although, creating a bucket as part of the startup might be a bad thing anyway - as this can be done in the UI/config.

Reply:
updated with _ready as internal.